network/phonebook_test.go

```diff
@@ -202,6 +203,69 @@ func TestMultiPhonebookDuplicateFiltering(t *testing.T) {
 	testPhonebookUniform(t, set, mp, 3)
 }
 
+func TestWaitAndAddConnectionTime(t *testing.T) {
+
+	entries := make(phonebookEntries, 0)
+	addr1 := "addrABC"
+	addr2 := "addrXYZ"
+
+	// addr not in. Should be added
+	entries.waitAndAddConnectionTime(addr1, 3, 2)
+	phBookData := entries[addr1]
+	assert.Equal(t, 1, len(phBookData.recentConnectionTimes))
+
+	// add 1 second gap between the two requests
+	time.Sleep(1 * time.Second)
+
+	// add another value to addr
+	entries.waitAndAddConnectionTime(addr1, 3, 2)
+	phBookData = entries[addr1]
+	assert.Equal(t, 2, len(phBookData.recentConnectionTimes))
+
+	// wait for the time the first element should be removed
+	time.Sleep(1 * time.Second)
+
+	// the first time should be removed and a new one added
+	entries.waitAndAddConnectionTime(addr1, 3, 2)
+	phBookData = entries[addr1]
+	assert.Equal(t, 2, len(phBookData.recentConnectionTimes))
```

Comment:
You also want to verify the content of the entries time.
i.e. save the content of the *second* element in the array *before* calling the waitAndAdd... and make sure it's the first element after.